### PR TITLE
WIP: report syntax errors in the web page

### PR DESCRIPTION
**Pull request by nickskalkin**
Hi! Here is my implementation of this feature #6248 

It works for me, but I think that implementation may not be very good. Can you give me some advices?
Also I haven't written any tests and docs, as I think I will be asked to rewrite my solution...

### CODE REVIEW

**Comment by DirtyF** [comment]
Thanks for giving a shot at this ❤️ 

My 2 cents on tests and documentation: Writing tests is the best way to make sure your code does what he's supposed to do. You won't need to change them even if there's need for refactoring during the review, it will even be easier for you to refactor with the help of the tests. The same goes for documentation. 😄

**Comment by DirtyF** [comment]
I can't test that locally with jekyll's website: 

```

**Comment by nickskalkin** [comment]
@ashmaroli I've tested only Liquid errors. For example I've changed `{% include ... %}` to `{% failed_include .... %}`

Here is an example how it looks for Jekyll site:

screen shot 2018-05-01 at 22 45 34

Some notes:
I haven't run jekyll site with `rake` task, only by `bundle exec jekyll --livereload`. Mention `livereload`, as my code work ONLY when livereload is `true`.

**Comment by ashmaroli** [comment]
@nickskalkin I tested this out locally using Jekyll's `docs` folder by adding `{% foobar %}` to `_docs/code_of_conduct.md` and `_tutorials/orderofinterpretation.md`, and using `master` versions of `jekyll-mentions` and `jemoji` plugins.
 Here are my findings: *(all commands are from within the `docs` directory and prepended with `bundle exec`)*
  - When I run `jekyll build`, the build aborts with an exception from our `jekyll-mentions` plugin on `_docs/code_of_conduct.md`
  - When I run `jekyll serve`, the build aborts as above.
  - When I run `jekyll serve --livereload`, the following can be observed:
    - When I visit `/tutorials/orderofinterpretation/`, the error template is rendered &mdash; **your implementation works!**
    - When I visit `/docs/conduct/`, the error template is **not rendered** &mdash; *your implementation got overridden by Jekyll's `:post_render` hook called by `jekyll-mentions` plugin*

So, for an initial draft :+1:  :clap:
Good start!!

**Comment by ashmaroli** [comment]
And regarding the code, I recommend *memoizing* the methods that are called multiple times but do not change for given build session, especially `LiveReloadTemplate#template` and `ErrorPrinter#error_styles` so that they do not get executed 1000 times in a large site when a *layout for 1000 posts* contains a Liquid Syntax error

**Comment by nickskalkin** [comment]
@ashmaroli thanks for the feedback! I'll improve my code (about memoizing, docs and tests). But maybe I didn't get you right, is current behaviour OK? As far as I understand I need to fix this:

> When I visit /docs/conduct/, the error template is not rendered — your implementation got overridden by Jekyll's :post_render hook called by jekyll-mentions plugin

**Comment by ashmaroli** [comment]
> As far as I understand I need to fix this:

Yes, that's correct.

**Comment by ashmaroli** [comment]
One other thing you **have to consider:** 
*If the Liquid Syntax Error occurs in a layout or an include, the rendered error is misleading..*

I don't know if you can overcome that hurdle...

**Comment by pathawks** [comment]
@nickskalkin Please do _not_ spend any time memorizing variables or performing premature optimizations.

Write readable code, and if we find a bottleneck we will deal with it on a case-by-case basis 👍🏼

**Comment by nickskalkin** [comment]
There are some tests and fix of post render hook

File: test/test_live_reload_template.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+require "helper"
+
+class TestLiveReloadTemplate < JekyllUnitTest
+  context "rendering livereload script with different options" do
+    context "with port only" do
+      subject { LiveReloadTemplate.new("livereload_port" => "35729") }
```
Comment (ashmaroli) [inline test/test_live_reload_template.rb]: should this be `35_729` like its in other tests..?

Comment (nickskalkin) [reply test/test_live_reload_template.rb]: OK, I'll change!

**Pull request closed without merging**